modules/core/src/passes/layers-pass.ts

```diff
@@ -59,14 +59,15 @@ export default class LayersPass extends Pass {
   render(options: LayersPassRenderOptions): any {
     // @ts-expect-error TODO - assuming WebGL context
     const [width, height] = this.device.canvasContext.getDrawingBufferSize();
+    const clearCanvas = options.clearCanvas ?? true;
 
     const renderPass = this.device.beginRenderPass({
       framebuffer: options.target,
       parameters: {
         viewport: [0, 0, width, height]
       },
       // clear depth and color buffers, restoring transparency
-      clearColor: options.clearColor ?? (options.clearCanvas ? [0, 0, 0, 0] : false),
+      clearColor: options.clearColor ?? (clearCanvas ? [0, 0, 0, 0] : false),
       clearDepth: options.clearCanvas ? 1 : undefined
```

Comment:
Use fallback value?

Reply:
Oh! Depth is already being cleared, and I think the reason clarifies how this broke in the first place:

https://github.com/visgl/luma.gl/blob/3ff710ed1685088f6148ea1b2e6b4fb2cb585d5c/modules/webgl/src/adapter/resources/webgl-render-pass.ts#L125-L136

We're checking `props.clearFoo !== false` in WebGLRenderPass, so an undefined is equivalent to clearing by default. But we do `(clearCanvas ? [0, 0, 0, 0] : false)` above, turning an `undefined` into `false`.

So maybe I should just remove the default from LayersPass too, and make sure it passes through undefined if `clearColor` and `clearCanvas` were not specified?

Reply:
... alternatively, we could remove the `clearCanvas` prop, it isn't widely used in the codebase, and use `clearColor` directly instead.

Reply:
I suspect this breaks mapbox/google maps interleaving, as they specify `clearCanvas: false` and expect depth not be cleared. The mapbox example is currently broken pending @chrisgervang 's PR. Any reason `undefined` is used here? @Ibgreen

Reply:
Makes sense. I've pushed another change that I think handles this as expected, if a bit verbosely. But also happy to wait on clarification and/or mapbox example updates.

Reply:
It seems to me that we should define how we want luma.gl RenderPass props to work. 
- I believe this API went through a few iterations as WebGPU is quite different from WebGL here 
- (clear can only be done as part of beginning a new RenderPass),
- and they have "load" constants for preserving contents of previous scene (which is apparently more expensive than clearing)
- If we revise the API, We could argue that one has to specify `clearXYZ: ...` to actually get things cleared, i.e. `undefined` should be no clearing. Even though that default is (slightly) more expensive.

Reply:
I don't think we need to change anything in luma, but your clarification on performance impact is be very helpful. I think the right thing to do is explicitly setting the `clear*` behaviors here instead of falling through to luma's RenderPass defaults.

Reply:
If luma.gl's RenderPass didn't clear by default, we'd still use WebGL's `preserveDrawingBuffer=false` by default, correct? That's certainly fine with me, and perhaps what I'd expect given luma.gl's lower level of abstraction.

And then deck.gl's LayersPass would probably do the opposite — clear unless otherwise specified.

Reply:
Just updated the PR to specify clearColor and clearDepth explicitly in LayersPass.